File: backend/http/types/sourcecontrol/member_activity.go
package sourcecontrol

import (
	"time"

	"gorm.io/datatypes"
)

// GetMemberActivityRequest represents the request parameters for getting member activity
type GetMemberActivityRequest struct {
	StartDate string `form:"startDate" binding:"omitempty"`
	EndDate   string `form:"endDate" binding:"omitempty"`
}

// MemberActivity represents a single activity item in the timeline
type MemberActivity struct {
	ID               string                 `json:"id"`
	Type             string                 `json:"type"` // "pull_request", "pr_review", "pr_comment"
	Title            string                 `json:"title"`
	Description      string                 `json:"description,omitempty"`
	URL              string                 `json:"url,omitempty"`
	Repository       string                 `json:"repository,omitempty"`
	CreatedAt        time.Time              `json:"created_at"`
	Metadata         map[string]interface{} `json:"metadata,omitempty"`
	AuthorUsername   string                 `json:"author_username,omitempty"`
	PRTitle          string                 `json:"pr_title,omitempty"`           // For comments/reviews: the PR title
	PRAuthorUsername string                 `json:"pr_author_username,omitempty"` // For comments/reviews: the PR author
	PRMetrics        datatypes.JSON         `json:"pr_metrics,omitempty"`
}

// GetMemberActivityResponse represents the response for getting member activity
type GetMemberActivityResponse struct {
	Activities []MemberActivity `json:"activities"`
}

File: frontend/src/types/memberActivity.ts
export interface MemberActivity {
  id: string
  type: 'pull_request' | 'pr_review' | 'pr_comment'
  title: string
  description?: string
  url?: string
  repository?: string
  createdAt: string
  metadata?: Record<string, any>
  authorUsername?: string
  prTitle?: string          // For comments/reviews: the PR title
  prAuthorUsername?: string // For comments/reviews: the PR author
  prMetrics?: Record<string, any> // PR performance metrics
}

export interface GetMemberActivityResponse {
  activities: MemberActivity[]
}

export interface GetMemberActivityParams {
  startDate?: string // YYYY-MM-DD format
  endDate?: string   // YYYY-MM-DD format
} 